purchase_stock_bl_date/models/purchase.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    bl_date = fields.Date()
+
+    @api.multi
+    def action_view_invoice(self):
+        res = super(PurchaseOrder, self).action_view_invoice()
+        if res.get("context"):
+            res["context"].update(
+                default_date_invoice=self.bl_date,
+            )
+        return res
```

Comment:
Can we remove this for now?  From my point of view, the user should just input in the invoice date field the date on the invoice from the supplier, and there is no difference whether there is B/L date or not (not perfectly sure, but that is my assumption).

```suggestion
```